src/components/text-editor/prosemirror-adapter/prosemirror-adapter.tsx

```diff
@@ -410,9 +410,27 @@ export class ProsemirrorAdapter {
     };
 
     private handleLinkChange = (event: CustomEvent<EditorTextLink>) => {
-        this.link = event.detail;
+        const { href } = event.detail;
+        const normalizedHref = this.normalizeExternalLink(href);
+
+        this.updateLink({
+            ...event.detail,
+            href: normalizedHref,
+        });
+    };
+
+    private normalizeExternalLink = (href: string): string => {
+        if (isExternalLink(href) && !/^https?:\/\//i.test(href)) {
+            return `https://${href}`;
+        }
+
+        return href;
     };
 
+    private updateLink = debounce((value: EditorTextLink) => {
+        this.link = value;
+    }, DEBOUNCE_TIMEOUT);
```

Comment:
Because when the user enters the url, we have to wait a bit to normalize (adding https or whatever we decide) the url. Otherwise it changes while the consumer are writing